File: CS50x-pset-2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>

int count_letters(string s);
int count_words(string s);
int count_sentences(string s);

int main(void)
{
    string s = get_string("Text: "); //input reading passage from user

    int letters = count_letters(s); //find number of letters
    int words = count_words(s); //find number of words
    int sentences = count_sentences(s); //find number of sentences

    //calculate grade (index)
    float grade = 0.0588 * ((float)100 * letters / words) - 0.296 * ((float)100 * sentences / words) - 15.8;

    //Output
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", grade);
    }
}

int count_letters(string s)
{
    //initialization
    int letters = 0;

    //count letters by using ascii codes
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if ((s[i] >= 65 && s[i] <= 90) || (s[i] >= 97 && s[i] <= 122))
        {
            letters++;
        }
    }

    //return number of letters
    return letters;
}

int count_words(string s)
{
    //initialization
    int words = 1;

    //count words - Ascii code for space = 32
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == 32)
        {
            words++;
        }
    }

    //return number of words
    return words;
}

int count_sentences(string s)
{
    //initialization
    int sentences = 0;

    //count sentences each time there is '.', '!', or '?'
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == 46 || s[i] == 63 || s[i] == 33)
        {
            sentences++;
        }
    }

    //return number of sentences
    return sentences;
}

File: CS50x-pset-6/readability/readability.py
def main():
    s = input("Text: ")  # input reading passage from user

    letters = count_letters(s)  # find number of letters
    words = count_words(s)  # find number of words
    sentences = count_sentences(s)  # find number of sentences

    # calculate grade (index)
    grade = 0.0588 * 100 * letters / words - 0.296 * 100 * sentences / words - 15.8

    # output
    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print("Grade", round(grade))


def count_letters(s):
    letters = 0
    # count letters by using ascii codes
    for c in s:
        c = ord(c)
        if (c >= 65 and c <= 90) or (c >= 97 and c <= 122):
            letters += 1

    # return number of letters
    return letters


def count_words(s):
    words = 1
    # count words - Ascii code for space = 32
    for c in s:
        c = ord(c)
        if c == 32:
            words += 1

    # return number of words
    return words
    

def count_sentences(s):
    sentences = 0
    # count sentences each time there is '.', '!', or '?'
    for c in s:
        c = ord(c)
        if (c == 46 or c == 63 or c == 33):
            sentences += 1

    # return number of sentences
    return sentences


main()